pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -719,3 +719,15 @@ func (p *provider) Validate(spec v1alpha1.MachineSpec) error {
 
 	return nil
 }
+
+func (p *provider) MachineMetricsLabels(machine *v1alpha1.Machine) map[string]string {
+	labels := make(map[string]string)
+
+	c, _, err := p.getConfig(machine.Spec.ProviderConfig)
+	if err == nil {
+		labels["size"] = c.VMSize
```

Comment:
> The "size" label must not have different names depending on the provider because otherwise we have to build a dashboard per provider.

Do we want to reduce the information to *only* machine type and provider? In that case I would change the `MachineMetricsLabels` function to something like `MachineTypeForMetrics`. Personally I would still prefer to have additional, provider-specific labels on the metrics.

> Also, errors here will go completely unnoticed except for the missing labels.

Propagating them would be hard, so let's just report them as runtime errors (`runtime.HandleError`)?

Reply:
> I would prefer not to make the call here.

Same. I will postpone any changes (explicit check for a given "size" label vs. having only a `GetInstanceType()` on the Provider interface vs. whatever else) until we're sure what labels we want to report.

All other things should be fixed by now. Not 100% if the vsphere implementation to create an instance type string is perfect, though.

Reply:
As Matze has now left the building, we need to stand on our feet :( @mrIncompetent what do you think? Keep the provider specific labels or remove them?

Reply:
I would try to keep the label names equal wherever possible as it makes building Dashboards easier.

IMO introducing provider-specific is fine - At least from my perspective, that's the only interesting information this controller can provide. The corev1.Node has already a lot of information about the host itself and we should not duplicate it.

The point regarding store-efficiency in Prometheus from @alvaroaleman should only hurt in mixed clusters: https://prometheus.io/docs/practices/naming/#labels
Those have not been our focus.

Leaving the decision up to @alvaroaleman

Reply:
Lets keep them then.